lint_rules/lint_rules/ifs_coding_standards_2011.py

```diff
@@ -486,20 +486,39 @@ def check_subroutine(cls, subroutine, rule_report, config, **kwargs):
             root_expr_map = defaultdict(list)
             for expr in expr_list:
                 expr_root = FindExpressionRoot(expr).visit(node)[0]
-                if expr_root.source and expr_root.source.string:
+                if node.source and node.source.string:
                     # Include only if we have source string information for this node
                     root_expr_map[expr_root] += [expr]
 
             # Then we look at the comparison operators for each expression root and match
             # them directly in the source string
             for expr_root, exprs in root_expr_map.items():
+                # find source lines for expression root
+                lstart, lend = node.source.find(str(expr_root))
+                lines = node.source.clone_lines((lstart, lend))
+
                 # For each comparison operator, check if F90 or F77 operators are matched
                 for op in sorted({op.operator for op in exprs}):
-                    source_string = strip_inline_comments(expr_root.source.string)
+
+                    # find source line for operator
+                    op_str = op if op != '!=' else '/='
+                    line = [line for line in lines if op_str in strip_inline_comments(line.string)]
+                    if not line:
+                        _op_map = {
```

Comment:
Just curious: any particular reason for moving the map here?

Reply:
As explained in #118, this ended up here because I misunderstood one of your earlier comments 😅